.NET/Microsoft.Recognizers.Text.DateTime/Chinese/Parsers/ChineseDatePeriodParserConfiguration.cs

```diff
@@ -676,6 +676,13 @@ private DateTimeResolutionResult ParseOneWordPeriod(string text, DateObject refe
                         swift = -1;
                     }
 
+                    // Handle cases with "(上|下)半" like "上半月"、 "下半年"
+                    if (!string.IsNullOrEmpty(match.Groups["halfTag"].Value))
+                    {
+                        ret = HandleWithHalfTag(trimmedText, referenceDate, ret, swift);
+                        return ret;
+                    }
+
                     if (trimmedText.EndsWith("周") | trimmedText.EndsWith("星期"))

```

Comment:
Such terms shouldn't be hardcoded. Can you move all to the YAML file?
Other YAML files have examples of this.